### ISSUE TITLE
Edge rendering occasionally crashes program

### ISSUE THREAD

**Issue by Sparen** [Sparen/Summer2016_Network#28]
At line 359 of the draw program, the following code is present:

```
// horizontally aligned
                    if (secondUDside === SAME) {
                        if (secondLRside === LEFT) {
                            arcTangentPoint = [currPoint[0] + curveRadius / UNIT, currPoint[1]];
                            nextPoint = [nextPoint[0] - curveRadius / UNIT, nextPoint[1]];
                        } else {
                            arcTangentPoint = [currPoint[0] - curveRadius / UNIT, currPoint[1]];
                            nextPoint = [nextPoint[0] + curveRadius / UNIT, nextPoint[1]];
                        }
                    }
```

There is no else statement, so it is possible for arcTangentPoint to be undefined immediately afterwards, crashing the program and preventing rendering of future objects.

This error seems to be triggered when an edge to a special type of plus point (containing three points total) is present.

Say we have edge X, with midpoints A (source), B (source stub), and C (sourcestubx, sourcestuby - 1.5). This type of edge must not crash the program.

**Comment by Sparen** [comment]
@alexahn917 Please consider adding the else statement as soon as possible so that our program no longer crashes occasionally.

**Comment by alexahn917** [comment]
I have worked some hours on figuring out this problem.
I was able to handle one of the missed out cases : horizontal alignment for the very first coordinate after stub point is now handled.
I am not sure how to handle this 'special case' which is at the 'else' statement at bottom.

**Comment by Sparen** [comment]
Your commit seems to have some strange side effects:
<img width="361" alt="screen shot 2016-09-20 at 5 19 26 pm" src="https://cloud.githubusercontent.com/assets/5430590/18689310/79806c86-7f56-11e6-8ae0-f1cec4bcaec3.png">

**Comment by Sparen** [comment]
Please address the issue, or the only solution will involve reverting commits/hackish stuff to prevent unusual behavior.

**Comment by alexahn917** [comment]
I will work on it in the morning

2016-09-24 0:26 GMT-04:00 Andrew Fan notifications@github.com:

**Comment by Sparen** [comment]
Alternate solution implemented as of b658932
